go/go116/Dockerfile

```diff
@@ -14,8 +14,24 @@
 
 FROM golang:1.16
 
-# Install python3 for trampoline.
-RUN apt-get update && apt-get install -y python3 python3-pip && which python3
+# Install pyenv
+RUN apt-get update
+RUN apt-get install -y --force-yes make build-essential libssl-dev zlib1g-dev libbz2-dev \
+    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+    xz-utils tk-dev libffi-dev liblzma-dev python-openssl
+RUN git clone https://github.com/pyenv/pyenv.git ~/.pyenv
```

Comment:
Maybe do something similar to https://github.com/googleapis/testing-infra-docker/blob/master/python/cloud-devrel-kokoro-resources/python-base/Dockerfile#L100?

Reply:
Sure. IT sounds like we prefer to install from source rather than use pyenv? Open to either option. 

Do we want to install specific/multiple python versions or just the latest (3.8) is ok? @codyoss

Reply:
I think for the Go image one version should be fine. @busunkim96 might have other recommendations on how to install, I know very little on the best way to do things in the Python ecosystem.

Reply:
pyenv seems to work well in the NodeJS and Java containers. https://github.com/googleapis/testing-infra-docker/search?q=pyenv 

The Python image installs from source only because it was originally set up that way and no one has changed it. 😅

Reply:
@codyoss , do I need to stage the image somewhere in order to pass go-presubmit? How might I best get this PR in? thank you!

Reply:
added the comment to kick off the build.